airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from "classnames";
+import { useCallback, useRef, useState } from "react";
+import React from "react";
+import { useInView } from "react-intersection-observer";
+import { useResizeDetector } from "react-resize-detector";
+
+import { Tooltip } from "components/ui/Tooltip";
+
+import { Text, TextProps } from "./Text";
+import styles from "./TextWithOverflowTooltip.module.scss";
+
+interface TextWithOverflowTooltipProps extends TextProps {
+  onlyCheckForSizeIfInView?: boolean;
+}
+export const TextWithOverflowTooltip: React.FC<React.PropsWithChildren<TextWithOverflowTooltipProps>> = React.memo(
+  ({ children, onlyCheckForSizeIfInView = false, className, ...props }) => {
+    const [tooltipDisabled, setTooltipDisabled] = useState(true);
+
+    const textRef = useRef<HTMLElement | null>(null);
+    const { inView, ref: inViewRef } = useInView({ delay: 1000, skip: !onlyCheckForSizeIfInView });
+
+    const onResize = useCallback(() => {
+      if (!onlyCheckForSizeIfInView || inView) {
+        setTooltipDisabled((textRef.current?.scrollWidth ?? 0) <= (textRef.current?.clientWidth ?? 0));
+      }
+    }, [inView, onlyCheckForSizeIfInView]);
+
+    const { ref: resizeRef } = useResizeDetector<HTMLElement>({
+      onResize,
+      refreshMode: "debounce",
+      handleHeight: false,
+      handleWidth: !onlyCheckForSizeIfInView || inView,
+    });
+
+    return (
+      <Tooltip
+        control={
+          <Text
+            ref={(element) => {
+              textRef.current = element;
+              resizeRef.current = element;
+              inViewRef(element);
+            }}
+            className={classNames(className, styles.overflowText)}
+            {...props}
+          >
+            {children}
+          </Text>
+        }
+        disabled={tooltipDisabled}
+        className={styles.fullWidthTooltip} // todo: is that right?
```

Comment:
I'm unsure what logic you're porting from but this will allow the tooltip to be any width.

Reply:
I pulled this in from somewhere in the catalog tree table row.

Looking at this with fresh eyes... We probably _don't_ want this by default so the tooltip will wrap.

Reply:
I'd agree with that. I have a tickling in the back of my head that I applied some width-adjusting style in my table work. Lemme know if you'd like to pair to sort through this!

Reply:
It was for when things have very long strings with no word break ie:
![](https://files.slack.com/files-pri/T02TET08MDH-F04RJ3V6PTQ/screenshot_2023-02-28_at_4.03.24_pm.png)

added a `break-word` instead, which seems cleaner?:
![](https://files.slack.com/files-pri/T02TET08MDH-F04RQN1TA8J/screenshot_2023-02-28_at_4.05.47_pm.png)